File: backend/src/main/scala/domains/blog/model/UpdateBlogRequest.scala
package domains.blog.model

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UpdateBlogRequest(
  title: BlogTitle,
  content: BlogContent,
  visibility: BlogVisibility
)

object UpdateBlogRequest:
  given Encoder[UpdateBlogRequest] = deriveEncoder[UpdateBlogRequest]
  given Decoder[UpdateBlogRequest] = deriveDecoder[UpdateBlogRequest]

File: frontend/src/features/blog/http/request/UpdateBlogRequest.ts
import type { BlogContent } from '@/features/blog/model/BlogContent'
import type { BlogTitle } from '@/features/blog/model/BlogTitle'
import type { BlogVisibility } from '@/features/blog/model/BlogVisibility'

export type UpdateBlogRequest = {
  title: BlogTitle
  content: BlogContent
  visibility: BlogVisibility
}
